src/storage/src/source/kafka.rs

```diff
@@ -401,34 +433,33 @@ impl SourceReader for KafkaSourceReader {
 }
 
 #[async_trait::async_trait]
-impl OffsetCommitter for KafkaOffsetCommiter {
+impl OffsetCommitter<Partitioned<i32, MzOffset>> for KafkaOffsetCommiter {
     async fn commit_offsets(
         &self,
-        offsets: HashMap<PartitionId, MzOffset>,
+        frontier: Antichain<Partitioned<i32, MzOffset>>,
     ) -> Result<(), anyhow::Error> {
         use rdkafka::consumer::CommitMode;
         use rdkafka::topic_partition_list::Offset;
 
+        // Generate a list of partitions that this worker is responsible for
+        let mut offsets = vec![];
+        for ts in frontier.iter() {
+            if let Some(pid) = ts.partition() {
```

Comment:
> This precludes our ability to consolidate Point's that are next to each other into Range's, without first fixing this, so this deserves a comment!

This is actually impossible. Points are never next to each other in the order defined by `Partitioned`, there is always a gap. So even if you had the frontier `{Point(1), Point(2)}` you wouldn't be able to merge it into `{Range(0, 3)}`. Even single points can't be transformed (e.g it's impossible to transform `{Point(1)}` to `{Range(0, 2)}`. This would amount to the frontier going backwards which isn't allowed